cmd/entrypoint/runner.go

```diff
@@ -19,7 +22,7 @@ type realRunner struct {
 
 var _ entrypoint.Runner = (*realRunner)(nil)
 
-func (rr *realRunner) Run(args ...string) error {
+func (rr *realRunner) Run(timeout string, args ...string) error {
```

Comment:
I might suggest this method taking a ctx argument instead of a string timeout, or if there's some reason not to take a ctx, have it take a time.Duration instead.

The reason is that because of the variadic arg, it'll be a bit confusing to see it called like `Run("30s", "echo", "hello")`, compared to `Run(ctx, "echo", "hello")` or slightly worse,`Run(30*time.Second, "echo", "hello")`

Reply:
Originally I did pass the context here, the reason that I'm now passing the timeout argument is that as soon as you add a deadline to a context, time starts running. The closer to the execution of the actual command the timeout is added to the context, the less timeout time is wasted on overhead before the command executes.

In reality this overhead (which technically could vary between systems) is probably in the order of milliseconds which may not be significant for the timeouts we expect to be set for a `Step`. If we accept that "wasting of time" it could also be done as you suggest.

Reply:
I think that's an acceptable skew from enforced timeout to "actual" timeout. If a user notices their timeouts being enforced 100 microseconds earlier, I'd be surprised :)